packages/istanbul-lib-coverage/lib/file-coverage.js

```diff
@@ -166,24 +207,37 @@ class FileCoverage {
             return;
         }
 
-        Object.entries(other.s).forEach(([k, v]) => {
-            this.data.s[k] += v;
-        });
-        Object.entries(other.f).forEach(([k, v]) => {
-            this.data.f[k] += v;
-        });
-        Object.entries(other.b).forEach(([k, v]) => {
-            let i;
-            const retArray = this.data.b[k];
-            /* istanbul ignore if: is this even possible? */
-            if (!retArray) {
-                this.data.b[k] = v;
-                return;
-            }
-            for (i = 0; i < retArray.length; i += 1) {
-                retArray[i] += v[i];
-            }
-        });
+        let [hits, map] = mergeProp(
+            this.s,
+            this.statementMap,
+            other.s,
+            other.statementMap
+        );
+        this.data.s = hits;
+        this.data.statementMap = map;
+
+        const keyFromLocProp = x => keyFromLoc(x.loc);
+        const keyFromLocationsProp = x => keyFromLoc(x.locations[0]);
+
+        [hits, map] = mergeProp(
```

Comment:
One slight concern I have about this change is the number of additional operations that will be performed on each coverage report as we iterate over all these branch coverage operations.

Maybe there's no way around it, if this is necessary for correct merging behavior, but it would be good to find a large codebase to test against.

Reply:
@bcoe The previous incarnation looped through each of these properties, including the branch hit counters, for just the "other" object. This new algorithm also needs to iterate through the properties for "this" object too. So in the worst case I would expect this to be potentially twice as slow, but that's a low price to pay for correctness.

Reply:
@JustinChristensen I'm pretty sure on large codebases using `c8`, it won't ultimately merge many coverage files because the merge happens at prior to processing when the output is in `v8` format.

I suppose the concern is a large codebase using `tap` and `nyc`, the `npm` CLI might be a good repo to test with -- if it works with npm, I'd be pretty confident we don't have too huge of a performance hit.